src/commands/Command.ts

```diff
@@ -0,0 +1,222 @@
+        const msg = alias ? `${this.optionAliasMessage} --${option.name}` : getValue(option.message) || ''
+        const env = alias ? alias.env : option.env
+        return (msg + (env ? ` (env: ${env})` : '')).split('\n').reduce((acc, l) => !acc ? l.trim() : `${acc}\n${spaces}${l.trim()}`, '')
+    }
+
+    private getOptions(params?: IParams): IOption[] {
+        return getValue(this.options, params) || []
+    }
+
+    protected param(params: IParams | undefined, name: string): IParam | undefined {
+        return params && params[name]
+    }
+
+    protected paramValue<T>(params: IParams | undefined, name: string): T | undefined {
+        return params && params[name] && params[name].value
+    }
+
+    protected paramFrom(params: IParams | undefined, name: string): ParamType | undefined {
+        return params && params[name] && params[name].from
+    }
+
+    protected getDefaultConfigFileOption(tap?: (param?: IParam) => void): IOption {
+        return {
+            name: this.configFileOptionName,
+            char: 'c',
+            env: 'CAPROVER_CONFIG_FILE',
+            message: 'path of the file where all parameters are defined in JSON or YAML format\n' +
+                     'see others options to know config file parameters\' names\n' +
+                     'this is mainly for automation purposes, see docs',
+            tap
+        }
+    }
+
+    public build() {
+        if (!this.command) throw 'Empty command name';
+
+        const cmd = this.program.command(this.command)
+        if (this.aliases && this.aliases.length) this.aliases.forEach(alias => alias && cmd.alias(alias))
+        if (this.description) cmd.description(this.description)
+        if (this.usage) cmd.usage(this.usage)
+
+        let options = this.getOptions().filter(opt => opt && opt.name)
+        const optionAliases: (IOptionAlias & {aliasTo: string})[] = options.reduce((acc, opt) => [...acc, {...opt, aliasTo: opt.name}, ...(opt.aliases || []).filter(alias => alias && alias.name).map(alias => ({...alias, aliasTo: opt.name}))], [])
+
+        options = options.filter(opt => !opt.hide)
+        const spaces = ' '.repeat(options.reduce((max, opt) => Math.max(max, this.getCmdLineFlags(opt, opt.type).length, (opt.aliases || []).filter(alias => alias && alias.name && !alias.hide).reduce((amax, a) => Math.max(amax, this.getCmdLineFlags(a, opt.type).length), 0)), 0) + 4)
+        options.forEach(opt => {
+            cmd.option(this.getCmdLineFlags(opt, opt.type), this.getCmdLineDescription(opt, spaces), getValue(opt.default))
+            if (opt.aliases) opt.aliases.filter(alias => alias && alias.name && !alias.hide).forEach(alias => cmd.option(this.getCmdLineFlags(alias, opt.type), this.getCmdLineDescription(opt, spaces, alias)))
+        })
+
+        cmd.action(async (cmdLineOptions: ICommandLineOptions) => {
+            cmdLineOptions = await this.preAction(cmdLineOptions)
```

Comment:
@Hazzard17h 

Unfortunately, I only found out about this after I release the new version.


This, `cmdLineOptions: ICommandLineOptions`, seems to be wrong. The argument to the action method is not an array of options. It's the whole Command object. I believe what you wanted to use is `cmdLineOptions.options` which translate to an object that has simple (key:string) => (value: string | boolean) structure.

Would you be able to fix the issue? Otherwise, let me know and I'll look into it. From what I can see, it's only being used to override the actual  parameters with the new aliases that you've added. Right?

Reply:
PS: Another issue in the same pipeline is that running an unknown command causes a problem. For example `caprover deploy random`

Reply:
As you can see [here](https://github.com/tj/commander.js/tree/v2.20.0#examples), the action of a program sub-command takes a function which has as many arguments as the positional parameters of the sub-command, and at least the entire sub-command object, with the options added as keys.
So, the code works if called normally, but as you had noted, if we pass a positional argument to a sub-command (e.g. `caprover deploy asd`), it is broken.
I'm fixing it, by trowing an error, if user provided a position parameter.

Reply:
See #23